presto-spark/src/main/java/com/facebook/presto/spark/planner/SparkPlanPreparer.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.planner;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.execution.Lifespan;
+import com.facebook.presto.execution.ScheduledSplit;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.scheduler.SqlQueryScheduler;
+import com.facebook.presto.execution.scheduler.SqlQueryScheduler.StreamingSubPlan;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.Split;
+import com.facebook.presto.operator.StageExecutionDescriptor;
+import com.facebook.presto.operator.StageExecutionDescriptor.StageExecutionStrategy;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.split.SplitManager;
+import com.facebook.presto.split.SplitSource;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.SubPlan;
+import com.google.common.collect.ImmutableSet;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.airlift.concurrent.MoreFutures.getFutureValue;
+import static com.facebook.presto.execution.scheduler.SqlQueryScheduler.extractStreamingSections;
+import static com.facebook.presto.execution.scheduler.TableWriteInfo.createTableWriteInfo;
+import static com.facebook.presto.operator.StageExecutionDescriptor.StageExecutionStrategy.UNGROUPED_EXECUTION;
+import static com.facebook.presto.spi.connector.ConnectorSplitManager.SplitSchedulingStrategy.UNGROUPED_SCHEDULING;
+import static com.facebook.presto.spi.connector.NotPartitionedPartitionHandle.NOT_PARTITIONED;
+import static com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class SparkPlanPreparer
```

Comment:
the name `SparkPlanPreparer` is a bit abstract, especially its relationship with `SparkQueryPlanner`  is not clear.  All the names might need a bit thought.

`SparkPlanFragmenter`, `SparkQueryPlanner` might just want to rename to `PrestoPlanFragmenter` and `PrestoQueryPlanner` since they are doing fragmenting queries for Presto, not Spark. The name looks like it's planning or fragmenting "Spark" query :) 

Maybe we should distinguish between a Presto plan and a "PrestoOnSpark" plan? , so:

- `PreparedPlan` -> `PrestoOnSparkPlan`
- `SparkPlanPreparer` -> `PrestoOnSparkPlanner` or `PrestoOnSparkPlanPreparer`, and add comment it basically takes a Presto plan and prepare a "PrestoOnSpark" Plan